src/Http/Routing/src/HttpMethodMetadata.cs

```diff
@@ -47,7 +47,7 @@ public HttpMethodMetadata(IEnumerable<string> httpMethods, bool acceptCorsPrefli
     /// <summary>
     /// Returns a value indicating whether the associated endpoint should accept CORS preflight requests.
     /// </summary>
-    public bool AcceptCorsPreflight { get; }
+    public bool AcceptCorsPreflight { get; internal set; }
```

Comment:
internal? 😢

Reply:
I'm sad about this too. The other option @brunolins16 showed me first was to look for the last `IHttpMethodMetadata` no matter what the type. And then add a new `HttpMethodMetadata` with the methods copied and `AcceptCorsPreflight` set to true. I thought this was worse though, because now the custom `IHttpMethodMetadata` type is no longer *the* `IHttpMethodMetadata`.

Can we add a default setter implementation to the interface that throws? That at least in theory would be extensible.

Reply:
So, will it require an API change, right? Let me work on it to have it ready for next API review.

Reply:
Is the suggestion to look for the last `IHttpMethodMetadata`, instead of the `HttpMethodMetadata`? If so, that means a custom `IHttpMethodMetadata` by default will throw when we set it in the new method, is that what you are saying?

Reply:
Changing to add a default implementation (we will need add to both setter and getter :( ) make the property hidden for custom implementations, no hint at all in the intellisense. To be honest, I am not sure if that is good.☹️

Reply:
> The other option @brunolins16 showed me first was to look for the last IHttpMethodMetadata no matter what the type. And then add a new HttpMethodMetadata with the methods copied and AcceptCorsPreflight set to true. I thought this was worse though, because now the custom IHttpMethodMetadata type is no longer the IHttpMethodMetadata.

I know @halter73 does not like it but I still prefer it.  I know that it could potentially break someone using `endpoint.Metadata.GetMetadata<IHttpMethodMetadata> as MyCustomMetadata`, however, expecting that the most significant metadata is the custom type is risky already because there is no guarantee that any other code added a metadata after it.

Reply:
> Is the suggestion to look for the last `IHttpMethodMetadata`, instead of the `HttpMethodMetadata`? If so, that means a custom `IHttpMethodMetadata` by default will throw when we set it in the new method, is that what you are saying?

Yes. I think the intersection of CORS and custom IHttpMethodMetadata is small, and we can make a clear exception message that says that the setter needs to be implemented.

> Changing to add a default implementation (we will need add to both setter and getter :( ) make the property hidden for custom implementations, no hint at all in the intellisense. To be honest, I am not sure if that is good.☹️

IHttpMethodMetadata.AcceptCorsPreflight is not an API that many ASP.NET Core developers are looking at with intellisense. It's an advanced API. As long as the exception is clear, I think we're fine.